src/Microsoft.Diagnostics.Monitoring.WebApi/Operation/EgressOperationStore.cs

```diff
@@ -121,13 +121,43 @@ public void CompleteOperation(Guid operationId, ExecutionResult<EgressResult> re
             }
         }
 
-        public IEnumerable<Models.OperationSummary> GetOperations()
+        public IEnumerable<Models.OperationSummary> GetOperations(ProcessKey? processKey)
         {
             lock (_requests)
             {
                 return _requests.Select((kvp) =>
                 {
                     EgressProcessInfo processInfo = kvp.Value.EgressRequest.EgressOperation.ProcessInfo;
+
+                    if (processKey != null)
```

Comment:
I think this makes more sense as a `Where` LINQ expression before this `Select` LINQ expression so that way we can avoid the need to return null and a `Where` after the `Select`. You could also conditionally construct the `IEnumerable` such that you don't have to check if the process key was provided for each iteration:

```cs
IEnumerable<KeyValuePair<Guid, EgressEntry>> requests = _requests;
if (null != processKey)
{
    requests = requests.Where(...);
}
return requets.Select(...).ToList();
```